Rerun fusion filtering with code modification + v11 data (6/N)

runjin326 opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?
Modify the module to be able to take `histologies-base.tsv` when running with `OPENPBTA_BASE_RELEASE=1 bash run_fusion_merged.sh`


#### What was your approach?
Made the following changes:
1) Only run script 06 when running with full histology 
2) Modified 04 and 05 scripts to take grouping variables depending on whether the run is for base histology or full histology file.
3) Modified run script so that the conditions can be run correctly


#### What GitHub issue does your pull request address?
https://github.com/PediatricOpenTargets/ticket-tracker/issues/316


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?
Please check to see whether using the grouping variables make sense
This module can be run by:
```
OPENPBTA_BASE_SUBTYPING=1 bash run_fusion_merged.sh 
```

#### Is there anything that you want to discuss further?
Note that for base run, I set the script to skip 06 so we do not have info about that yet. 

The reason for doing this is because currently, we are saying that toolkit will run this and we will run independent samples + fusion summary after toolkit runs is completed. So when toolkit runs this module, we would not have independent samples just yet. So what we have two option:
1) Skip 06 when running with base histology (as is committed here) 
2) Have toolkit run independent samples as well - and run this module AFTER independent samples module

So for this PR, we can:
1) Merge as is and re-run on full histology to QC using script 06; OR
2) Merge independent samples first and then run 06 to QC

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes.


### Results

#### What types of results are included (e.g., table, figure)?
```
FilteredFusion.tsv
fusion-putative-oncogenic.tsv
```


#### What is your summary of the results?
See discussion further above.


#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

jharenza commented on analyses/fusion_filtering/05-QC_putative_onco_fusion_distribution.Rmd:
```diff
@@ -131,8 +143,8 @@ putative_driver_annotated_fusions_local<-fusion_calls %>%
 plotNcg<-function(standardFusionCalls,filterN,plot_title){
-  # get count of number of cancer_group per FusionName
-  plotNcg<-standardFusionCalls %>% dplyr::select(FusionName,cancer_group) %>% unique() %>%
+  # get count of number of grouping varaible of full histology per FusionName
```
```suggestion
  # get count of number of grouping variable of full histology per FusionName
```

jharenza reviewed (commented):
I think we should do 
>Merge independent samples first and then run 06 to QC

jharenza commented:
So far ran fine and got the same output

jharenza commented:
Note to self: Update to remove `[],` annotation in putative oncogenic file per https://github.com/PediatricOpenTargets/ticket-tracker/issues/359

runjin326 commented:
@jharenza - this is ready for review. Noting a couple things here:
1) I merged `dev` since I need the independent sample list files but `gsea-v11` branch does not have dev merged yet. Once `gsea-v11` has dev merged, the additional changes from independent samples would not show.
2) I also modified the two generate file scripts + run subtyping script.
    - We need to run independent sample module with `OPENPBTA_BASE_SUBTYPING=1` for toolkit before running fusion filtering to avoid errors 
    - We do not need to run independent samples for subtyping script (I moved it to the generate files toolkit) - assuming we will generate all the files for releases (other than histology file) and have all those files ready before subtyping.
3) When I was fixing two generate file scripts, I realized an error in focal CN module, so I fixed that as well.
4) Also note that since fusion filtering will now be run with base histology, 06 is changed to group by broad histology instead of broad histology. 

Let me know if you have any question.

jharenza commented:
Can you remind me why we have `[run-prepare-cn-OpenTarget.sh](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/195/files#diff-c8da87933d704b6af050cde6147860ad2386e20714ae1221ebb89dbcdaeb0c92)` instead of just `run-prepare-cn`?

For the filenames and scripts, can you remove `by-broadhist` to keep them generalized?

jharenza commented on analyses/fusion_filtering/run_fusion_merged.sh:
```diff
@@ -46,11 +47,11 @@ normal_expression_adrenal_gland="${references_path}/gtex_adrenal_gland_TPM_hg38.
 # independent sample list
-independent_RNA_primary="${data_path}/independent-specimens.rnaseq.primary.tsv"
-independent_RNA_relapse="${data_path}/independent-specimens.rnaseq.relapse.tsv"
+independent_RNA_primary="${indep_path}/independent-specimens.rnaseq.primary-pre-release.tsv"
+independent_RNA_relapse="${indep_path}/independent-specimens.rnaseq.relapse-pre-release.tsv"
```
I think we might want to put these into v11 and still use the data folder to avoid module paths. Can you update those on s3 please?

runjin326 replied:
sure - done.

jharenza commented on scripts/generate-analysis-files-toolkit.sh:
```diff
@@ -25,10 +25,13 @@ scratch_dir="$BASEDIR/scratch"
 # Fusion filtering
 echo "Create fusion filtered list"
-OPENPBTA_BASE_RELEASE=1 bash ${analyses_dir}/fusion_filtering/run_fusion_merged.sh
+OPENPBTA_BASE_SUBTYPING=1 bash ${analyses_dir}/fusion_filtering/run_fusion_merged.sh
```
do you also want to add fusion summary to this script while you are editing it

runjin326 replied:
This already exists in `generate-analysis-files-repo.sh`

jharenza replied:
ok, I think I missed that

runjin326 commented:
> Can you remind me why we have `[run-prepare-cn-OpenTarget.sh](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/195/files#diff-c8da87933d704b6af050cde6147860ad2386e20714ae1221ebb89dbcdaeb0c92)` instead of just `run-prepare-cn`?
> 
> For the filenames and scripts, can you remove `by-broadhist` to keep them generalized?

I think at that point, Krutika mentioned maybe to keep it as reference? I am not 100% sure.

I changed by-broadhist to `bygroup` for file names and changed the script name to `-sample-group`.

runjin326 commented:
@jharenza - this is ready for another look.
